test/user_test.dart

```diff
@@ -0,0 +1,90 @@
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:io';
+
+import 'package:test/expect.dart';
+
+import '../lib/realm.dart';
+import 'test.dart';
+
+Future<void> main([List<String>? args]) async {
+  print("Current PID $pid");
+
+  await setupTests(args);
+
+  baasTest('User logout anon user is marked as removed', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.anonymous());
+    expect(user.state, UserState.loggedIn);
+    await user.logout();
+    expect(user.state, UserState.removed);
+  });
+
+  baasTest('User logout', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.emailPassword(testUsername, testPassword));
+    expect(user.state, UserState.loggedIn);
+    await user.logout();
+    expect(user.state, UserState.loggedOut);
+  });
+
+  baasTest('User get id', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.emailPassword(testUsername, testPassword));
+    expect(user.id, isNotEmpty);
+  });
+
+  baasTest('User get identities', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.emailPassword(testUsername, testPassword));
+    expect(user.identities.any((identity) => identity.provider == AuthProviderType.emailPassword), isTrue);
+  }, skip: "Blocked on https://github.com/realm/realm-core/issues/5467");
+
+  baasTest('User get customdata', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.emailPassword(testUsername, testPassword));
+    expect(user.customData, isNull);
+  });
+
+  baasTest('User refresh customdata', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.emailPassword(testUsername, testPassword));
+    final dynamic data = await user.refreshCustomData();
+    expect(data, isNull);
+  });
+
+  baasTest('User link credentials', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.anonymous());
+  
+    expect(user.state, UserState.loggedIn);
+    expect(user.identities.length, 1);
+    expect(user.identities.singleWhere((identity) => identity.provider == AuthProviderType.anonymous).provider, isNotNull);
+
+    final authProvider = EmailPasswordAuthProvider(app);
+    final username = "${generateRandomString(20)}@realm.io";
+    final password = generateRandomString(8);
+    await authProvider.registerUser(username, password);
+
+    await user.linkCredentials(Credentials.emailPassword(username, password));
+    expect(user.identities.length, 2);
+    expect(user.identities.singleWhere((identity) => identity.provider == AuthProviderType.emailPassword).provider, isNotNull);
+    expect(user.identities.singleWhere((identity) => identity.provider == AuthProviderType.anonymous).provider, isNotNull);
```

Comment:
We should further test that the user can now login with their email/password and that we get the same user as the one we already had.